### PR TITLE
pins: fix adding and removing pinned DMs & Groups

### PR DESCRIPTION
**Pull request by latter-bolden**
- Moves group pins from `channels` agent to `groups`
-  Defends against accidentally resetting pins since we don't update them individually on the backend
- Fixes groups pinning hooks that were unintentionally updated to dm hooks as part of the frontend state rewrite

Fixes LAND-1267

### CODE REVIEW

**Comment by arthyn** [comment]
Confused why we're removing the ability to pin channels

**Review by arthyn — COMMENTED** [review]
seems good just a few comments

File: ui/src/dms/MessagesSidebarItem.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ function ChannelSidebarItem({
         <DmOptions
           open={optionsOpen}
           onOpenChange={onOptionsOpenChange}
-          whom={whom}
+          whom={nest}
```
[the remark this answers is not in the document]
Comment (latter-bolden) [reply ui/src/dms/MessagesSidebarItem.tsx:61]: Should be a no-op — `nest` is instantiated a few lines up and gets set equal to `whom`. Just changed to express intent given how I'm using it in `DmOptions`.

Looks fine after merging your branch, but let me know if you see issues

File: desk/lib/groups-json.hoon [hunk 2]
```diff
@@ -330,16 +330,42 @@
       %chi  synced/~
       %lev  behind/~
     ==
+  ::
+  ++  whom
+    |=  w=whom:u
+    :-  %s
+    ?-  w
+      [%group *]       (flag flag.w)
```
[the remark this answers is not in the document]
Comment (latter-bolden) [reply desk/lib/groups-json.hoon:338]: Mark and I talked about this a bit, we'll have to adapt the response either way. 

Given the frontend parsing logic is well encapsulated and uses pre-existing helper functions, I think it's fine to leave as is.

**Comment by latter-bolden** [comment]
Should be ready to merge after @Fang- resolves the `talk` desk issues

**Pull request merged**